src/Sarif.Multitool/TransformCommand.cs

```diff
@@ -33,21 +45,32 @@ public static int Run(TransformOptions transformOptions)
                     : Formatting.None;
 
                 // Assume the input log is the "other" version
-                if (transformOptions.Version == 2)
+                if (transformOptions.Version == SarifVersion.Current)
                 {
-                    SarifLogVersionOne actualLog = FileHelpers.ReadSarifFile<SarifLogVersionOne>(transformOptions.InputFilePath, SarifContractResolverVersionOne.Instance);
-                    var visitor = new SarifVersionOneToCurrentVisitor();
-                    visitor.VisitSarifLogVersionOne(actualLog);
+                    string inputFilePath = transformOptions.InputFilePath;
+                    string inputVersion = SniffVersion(inputFilePath);
 
-                    FileHelpers.WriteSarifFile(visitor.SarifLog, fileName, formatting);
+                    if (inputFilePath == null || inputFilePath == "1.0.0")
+                    {
+                        SarifLogVersionOne actualLog = FileHelpers.ReadSarifFile<SarifLogVersionOne>(_fileSystem, transformOptions.InputFilePath, SarifContractResolverVersionOne.Instance);
+                        var visitor = new SarifVersionOneToCurrentVisitor();
+                        visitor.VisitSarifLogVersionOne(actualLog);
+                        FileHelpers.WriteSarifFile(_fileSystem, visitor.SarifLog, fileName, formatting);
+                    }
+                    else
+                    {
+                        // We have a pre-release v2 file that we should upgrade to current. 
+                        string sarifText = PrereleaseCompatibilityTransformer.UpdateToCurrentVersion(_fileSystem.ReadAllText(inputFilePath), formatting: formatting);
```

Comment:
>UpdateToCurrentVersion [](start = 78, length = 22)

Hmm. Elsewhere in this review I made a comment to the effect that perhaps you didn't want to read an entire potentially large file into memory. But we are doing it here.

Reply:
Right! This is a limit of JObject.Parse, which operates against a sting representation. I didn't want to exacerbate the situation further. The prerelease transformation is intended to wither away and it is already sufficiently burdensome to maintain this code. And so, performance is a lower priority. The goal, of course, is to finalize v2 and dispense with the rewrite altogether. In the future, maybe someone can go back and rewrite the transformation to be more performant. 

---
In reply to: [229514883](https://github.com/Microsoft/sarif-sdk/pull/1125#discussion_r229514883) [](ancestors = 229514883)